crates/sui/src/keytool.rs

```diff
@@ -878,53 +923,80 @@ impl KeyToolCommand {
+                            OIDCProvider::Slack,
+                            OIDCProvider::Apple,
+                            OIDCProvider::Kakao,
+                        ] {
+                            let mut e = fetch_jwks(&p, &client).await?;
+                            jwks.append(&mut e);
+                        }
                         let parsed: ImHashMap<JwkId, JWK> = jwks.clone().into_iter().collect();
-                        let aux_verify_data = VerifyParams::new(
-                            parsed,
-                            vec![
-                                OIDCProvider::Facebook,
-                                OIDCProvider::Twitch,
-                                OIDCProvider::Google,
-                            ],
-                            ZkLoginEnv::Prod,
-                            true,
-                        );
-                        let res = zk.verify_authenticator(
-                            &IntentMessage::new(Intent::sui_transaction(), tx_data.clone()),
-                            tx_data.execution_parts().1,
-                            Some(curr_epoch.unwrap()),
-                            &aux_verify_data,
-                        );
+                        let env = match network.as_str() {
+                            "devnet" | "localnet" => ZkLoginEnv::Test,
+                            "mainnet" | "testnet" => ZkLoginEnv::Prod,
+                            _ => return Err(anyhow!("Invalid network")),
+                        };
+                        let aux_verify_data = VerifyParams::new(parsed, vec![], env, true);
+
+                        let (serialized, res) = match IntentScope::try_from(intent_scope)
+                            .map_err(|_| anyhow!("Invalid scope"))?
+                        {
+                            IntentScope::TransactionData => {
+                                let tx_data: TransactionData = bcs::from_bytes(
+                                    &Base64::decode(&bytes.unwrap())
+                                        .map_err(|e| anyhow!("Invalid base64 tx data: {:?}", e))?,
+                                )?;
+
+                                let res = zk.verify_authenticator(
+                                    &IntentMessage::new(Intent::sui_transaction(), tx_data.clone()),
+                                    tx_data.execution_parts().1,
+                                    Some(curr_epoch.unwrap()),
+                                    &aux_verify_data,
+                                );
+                                (serde_json::to_string(&tx_data)?, res)
+                            }
+                            IntentScope::PersonalMessage => {
+                                let data: PersonalMessage = bcs::from_bytes(
+                                    &Base64::decode(&bytes.unwrap())
+                                        .map_err(|e| anyhow!("Invalid base64 tx data: {:?}", e))?,
```

Comment:
Is this tx data or an intent message?  
```suggestion
                                        .map_err(|e| anyhow!("Invalid base64 intent data: {:?}", e))?,
```